Network/NetworkService.swift

```diff
@@ -63,4 +72,34 @@ class NetworkService {
     func cancel() {
         task?.cancel()
     }
+    
+    
+    //MARK: Private
+    private func makeQuery(for url: URL, params: [String: AnyObject], type: QueryType) -> NSMutableURLRequest? {
```

Comment:
Later on you need to set header fields and http method so IMO it has to be NSMutableURLRequest.